addons/mail/models/discuss/discuss_channel.py

```diff
@@ -1073,13 +1107,56 @@ def create_group(self, partners_to, default_display_mode=False, name=''):
         channel._broadcast(channel.channel_member_ids.partner_id.ids)
         return channel
 
+    def create_sub_channel(self, initial_message_id=None, name=None):
+        self.ensure_one()
+        if self.owner_id:
+            raise ValidationError(_("You cannot create a sub channel from a sub-channel."))
+        if initial_message_id and name:
+            raise UserError(_("You cannot specify both initial_message_id and name."))
+        message = self.env["mail.message"]
+        if initial_message_id:
+            message = self.env["mail.message"].search([("id", "=", initial_message_id)])
+        sub_channel = self.create(
+            {
+                "channel_member_ids": [Command.create({"partner_id": self.env.user.partner_id.id})],
+                "channel_type": "channel",
+                "group_public_id": self.group_public_id.id,
+                "initial_message_id": message.id,
+                "name": message.body.striptags()[:30] if message else name or _("New Thread"),
+                "owner_id": self.id,
+            }
+        )
+        store = Store(sub_channel)
+        store.add(sub_channel, {"forceOpen": True})
+        self.env.user.partner_id._bus_send("mail.record/insert", store.get_result())
+        notification = (
+            Markup('<div class="o_mail_notification">%s</div>')
+            % _(
+                "%(user)s started a thread: %(goto)s%(thread_name)s%(end)s.%(goto_all)sSee all threads%(end)s."
+            )
+        ) % {
+            "user": self.env.user.display_name,
+            "goto": Markup(
+                "<a href='#' class='o_channel_redirect' data-oe-id='%s' data-oe-model='discuss.channel'>"
+            )
+            % sub_channel.id,
+            "goto_all": Markup("<a href='#' data-oe-type='sub-channels-menu'>"),
+            "thread_name": sub_channel.name,
+            "end": Markup("</a>"),
+        }
+        self.message_post(
+            body=notification, message_type="notification", subtype_xmlid="mail.mt_comment"
+        )
+        return store.get_result()
```

Comment:
The problem exists for every notification we send. I think it would be even weirder to get one translated correctly and the others in the sender's language. We should address this issue for all notifications at once, in a separate task IMO.